src/handlers.rs

```diff
@@ -131,3 +150,136 @@ pub extern "C" fn handle_code_cache(
   }
   // null response indicates success.
 }
+
+fn set_timeout<F>(
+  cb: F,
+  delay: u32,
+) -> (
+  impl Future<Item = (), Error = ()>,
+  futures::sync::oneshot::Sender<()>,
+)
+where
+  F: FnOnce() -> (),
+{
+  let (cancel_tx, cancel_rx) = oneshot::channel::<()>();
+  let when = Instant::now() + Duration::from_millis(delay.into());
+  let delay_task = Delay::new(when)
+    .map_err(|e| panic!("timer failed; err={:?}", e))
+    .and_then(|_| {
+      cb();
+      Ok(())
+    })
+    .select(cancel_rx)
+    .map(|_| ())
+    .map_err(|_| ());
+
+  (delay_task, cancel_tx)
+}
+
+fn set_interval<F>(
+  cb: F,
+  delay: u32,
+) -> (
+  impl Future<Item = (), Error = ()>,
+  futures::sync::oneshot::Sender<()>,
+)
+where
+  F: Fn() -> (),
+{
+  let (cancel_tx, cancel_rx) = oneshot::channel::<()>();
+  let when = Duration::from_millis(delay.into());
```

Comment:
This is misnamed - `when` suggests that this is an absolute time, but the variable actually contains a time delta.
If it had said `let when = Instant::now() + Duration::from_millis(delay.into())` it'd be accurate.

Reply:
Fixed.